### PR TITLE
Rule S6602: Update the message for arrays

### LINKED ISSUE #8935: Rule S6602: Improve the logging message when recommending for Array.Find
In the case of `Array.Find`, the recommendation is not very clear since it doesn't follow the pattern of the method already called; it's static. Make sure that the message is clear so the user understands how the method should be called.

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Fixes #8935

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Rules/FindInsteadOfFirstOrDefaultBase.cs [hunk 1]
```diff
@@ -48,7 +50,8 @@ public abstract class FindInsteadOfFirstOrDefaultBase<TSyntaxKind, TInvocationEx
                     && IsCorrectType(left, c.SemanticModel)
                     && !Language.Syntax.IsInExpressionTree(c.SemanticModel, invocation))
                 {
-                    c.ReportIssue(Diagnostic.Create(Rule, Language.Syntax.NodeIdentifier(invocation)?.GetLocation()));
+                    var message = c.SemanticModel.GetTypeInfo(left).Type.DerivesFrom(KnownType.System_Array) ? ArrayMessage : GenericMessage;
+                    c.ReportIssue(Diagnostic.Create(Rule, Language.Syntax.NodeIdentifier(invocation)?.GetLocation(), message));
```
[the remark this answers is not in the document]
Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/FindInsteadOfFirstOrDefaultBase.cs:54]: While your suggestion is more performant, I prefer to use the out parameters only if needed. The second call should be rather fast due to caching.

[the remark this answers is not in the document]
Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/FindInsteadOfFirstOrDefaultBase.cs]: It looks like the out parameter won 😱

**Review by gregory-paidis-sonarsource — APPROVED** [review]
LGTM

File: analyzers/src/SonarAnalyzer.Common/Rules/FindInsteadOfFirstOrDefaultBase.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (gregory-paidis-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/FindInsteadOfFirstOrDefaultBase.cs]: 😱

**Pull request merged**